### ISSUE TITLE
Breaking Changes since v1.3: Shipping Address component clean up (#3641)

### ISSUE THREAD

**Issue by Zeyber** [SAP/cloud-commerce-spartacus-storefront#3921]
Need to migrate shipping address component to cleaner component in #3641.

**`shipping-address.component.html`**

- Change all instances of `(click)="addressSelected(card.address)"` to `(click)="selectAddress(card.address)"`

- Change all instances of `(click)="back()"` to `(click)="goPrevious()"`

- Change all instances of `(click)="next()"` to `(click)="goNext()"`

- Change all instances of `[disabled]="!selectedAddress || !selectedAddress.id"` to `[disabled]="!(selectedAddress$ | async)?.id"`

- Change all instances of `(submitAddress)="addNewAddress($event)"` to `(submitAddress)="addAddress($event)"`

**`shipping-address-component.ts`**

- Remove deprecated method and variable calls from `ngOnInit()`

- Remove public variables and methods with deprecation notes

- Change class implements `export class ShippingAddressComponent implements OnInit, OnDestroy {` to `export class ShippingAddressComponent implements OnInit {`

**Comment by marlass** [comment]
Also remove unused services from constructor (eg. cartService)

**Comment by bgambocjaviniar** [comment]
QA results

works for both new user and already registered user

- can click back to go to cart and have the same state as it was before
- can submit shipping address to move on to the next checkout step
- can add new address
- can select an address to be the one used for checkout
- can checkout 

Everything is good, however missing a key piece for shipping address which will be addressed in https://github.com/SAP/spartacus/issues/7153
Basically, we can't set a new default address when we're in the checkout process
